### PR TITLE
[Improvement](set) change set_operator's hash map to phmap

### PR DESCRIPTION
**Pull request by BiteTheDDDDt**
### What problem does this PR solve?

1. change set_operator's hash map to phmap.
2. optimize the logic of refresh hash table 

```sql
+-------+--------------+------+-------+---------+-------+
| Field | Type         | Null | Key   | Default | Extra |
+-------+--------------+------+-------+---------+-------+
| k1    | int          | Yes  | true  | NULL    |       |
| k2    | int          | No   | false | NULL    | NONE  |
| k3    | bigint       | Yes  | false | NULL    | NONE  |
| k4    | varchar(100) | Yes  | false | NULL    | NONE  |
+-------+--------------+------+-------+---------+-------+
```

```sql
select count(*) from (select * from a_table intersect select * from  b_table)t;
before:Time(ms)=37768|CpuTimeMS=61754|PeakMemoryBytes=129921088
after: Time(ms)=25218|CpuTimeMS=44571|PeakMemoryBytes=129902624
```
```sql
select count(*) from (select k1 from a_table intersect select k1 from  b_table)t;
before: Time(ms)=7964|CpuTimeMS=13431|PeakMemoryBytes=130062400
after:  Time(ms)=7459|CpuTimeMS=11976|PeakMemoryBytes=130091048
```

Problem Summary:

### Check List (For Committer)

- Test 

    - [ ] Regression test
    - [ ] Unit Test
    - [ ] Manual test (add detailed scripts or steps below)
    - [ ] No need to test or manual test. Explain why:
        - [ ] This is a refactor/code format and no logic has been changed.
        - [x] Previous test can cover this change.
        - [ ] No colde files have been changed.
        - [ ] Other reason 

- Behavior changed:

    - [x] No.
    - [ ] Yes. 

- Does this need documentation?

    - [x] No.
    - [ ] Yes. 

- Release note

    
    
    None

### Check List (For Reviewer who merge this PR)

- [ ] Confirm the release note
- [ ] Confirm test cases
- [ ] Confirm document
- [ ] Add branch pick label

### CODE REVIEW

**Comment by BiteTheDDDDt** [comment]
run buildall

**Comment by BiteTheDDDDt** [comment]
run buildall

**Comment by BiteTheDDDDt** [comment]
run buildall

**Comment by BiteTheDDDDt** [comment]
run buildall

File: be/src/pipeline/exec/set_source_operator.cpp [hunk 1]
```diff
@@ -139,9 +138,21 @@ Status SetSourceOperatorX<is_intersect>::_get_data_in_hashtable(
...
+    };
+
+    auto& iter = hash_table_ctx.iterator;
+    for (; iter != hash_table_ctx.hash_table->end() && block_size < batch_size; ++iter) {
+        add_result(iter->get_second());
     }
 
     *eos = iter == hash_table_ctx.hash_table->end();
+    if (*eos && hash_table_ctx.hash_table->has_null_key_data()) {
+        auto value = hash_table_ctx.hash_table->template get_null_key_data<RowRefListWithFlags>();
```
Comment (HappenLee) [inline be/src/pipeline/exec/set_source_operator.cpp:150]: should here need copy? why here need check type?

Comment (BiteTheDDDDt) [reply be/src/pipeline/exec/set_source_operator.cpp:150]: has_null_key_data return true means hash table is `DataWithNullKey`, return type of other types of hash tables(except DataWithNullKey) is not RowRefListWithFlags

**Review by zhangstar333 — APPROVED** [review]

**Review by mrhhsg — APPROVED** [review]
LGTM

**Pull request merged**